pkg/cephmgr/rgw/bucket.go

```diff
@@ -49,53 +82,136 @@ func GetBucketStats(context *clusterd.Context, getClusterInfo func() (*mon.Clust
+	}
+
+	createdAt, err := time.Parse("2006-01-02 15:04:05.999999999Z", s.Data.CreationTime)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing date (%s): %+v", s.Data.CreationTime, err)
+	}
+
+	return &model.ObjectBucketMetadata{Owner: s.Data.Owner, CreatedAt: createdAt}, nil
+}
+
 func ListBuckets(context *clusterd.Context, getClusterInfo func() (*mon.ClusterInfo, error)) ([]model.ObjectBucket, error) {
 	logger.Infof("Listing buckets")
 
-	stats, err := GetBucketStats(context, getClusterInfo)
+	stats, err := GetBucketsStats(context, getClusterInfo)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to get bucket stats: %+v", err)
 	}
 
 	buckets := []model.ObjectBucket{}
 
 	for bucket, stat := range stats {
-		result, err := RunAdminCommand(context, getClusterInfo,
-			"metadata",
-			"get",
-			"bucket:"+bucket)
+		metadata, err := getBucketMetadata(context, bucket, getClusterInfo)
 		if err != nil {
-			return nil, fmt.Errorf("failed to list buckets: %+v", err)
+			return nil, err
 		}
 
-		var s struct {
-			Data struct {
-				Owner        string `json:"owner"`
-				CreationTime string `json:"creation_time"`
-			} `json:"data"`
-		}
-		if err := json.Unmarshal([]byte(result), &s); err != nil {
-			return nil, fmt.Errorf("failed to read buckets list. %+v, result=%s", err, result)
+		buckets = append(buckets, model.ObjectBucket{Name: bucket, ObjectBucketMetadata: model.ObjectBucketMetadata{Owner: metadata.Owner, CreatedAt: metadata.CreatedAt}, ObjectBucketStats: stat})
+	}
+
+	return buckets, nil
+}
+
+func GetBucket(context *clusterd.Context, bucket string, getClusterInfo func() (*mon.ClusterInfo, error)) (*model.ObjectBucket, int, error) {
+	stat, err := GetBucketStats(context, bucket, getClusterInfo)
+	if err != nil {
+		if err.Error() == "not found" {
+			return nil, RGWErrorNotFound, fmt.Errorf("Bucket not found")
 		}
+		return nil, RGWErrorUnknown, fmt.Errorf("Failed to get bucket stats: %+v", err)
+	}
 
-		createdAt, err := time.Parse("2006-01-02 15:04:05.999999999Z", s.Data.CreationTime)
-		if err != nil {
-			return nil, fmt.Errorf("Error parsing date (%s): %+v", s.Data.CreationTime, err)
+	metadata, err := getBucketMetadata(context, bucket, getClusterInfo)
+	if err != nil {
+		if err.Error() == "not found" {
+			return nil, RGWErrorNotFound, fmt.Errorf("Bucket not found")
 		}
+		return nil, RGWErrorUnknown, err
+	}
 
-		buckets = append(buckets, model.ObjectBucket{Name: bucket, Owner: s.Data.Owner, CreatedAt: createdAt, ObjectBucketStats: stat})
+	return &model.ObjectBucket{Name: bucket, ObjectBucketMetadata: model.ObjectBucketMetadata{Owner: metadata.Owner, CreatedAt: metadata.CreatedAt}, ObjectBucketStats: *stat}, RGWErrorNone, nil
+}
+
+// FIXME: The "bucket create" radowsgw-admin command does not exist yet
+func CreateBucket(context *clusterd.Context, bucket model.ObjectBucket, getClusterInfo func() (*mon.ClusterInfo, error)) (*model.ObjectBucket, error) {
```

Comment:
similar to ACLs, i don't think it makes sense to expose the create command without an implementation